src/azure-cli/azure/cli/command_modules/acr/connected_registry.py

```diff
@@ -394,20 +394,14 @@ def _get_install_info(cmd,
                       registry_name,
                       regenerate_credentials,
                       resource_group_name=None):
-    registry, resource_group_name = validate_managed_registry(
+    _, resource_group_name = validate_managed_registry(
         cmd, registry_name, resource_group_name)
     connected_registry = acr_connected_registry_show(
         cmd, client, connected_registry_name, registry_name, resource_group_name)
     parent_gateway_endpoint = connected_registry.parent.sync_properties.gateway_endpoint
-    parent_id = connected_registry.parent.id
+    if parent_gateway_endpoint is None or parent_gateway_endpoint == '':
+        parent_gateway_endpoint = "<parent gateway endpoint>"
```

Comment:
You may consider to add a variable 

// parent is cloud registry
parent_endpoint_protocol = "https"

// parent is connected registry
if parent_gateway_endpoint is None or parent_gateway_endpoint == '':
  parent_endpoint_protocol = "<http or https>"